src/plugins/impl/fingerprint.plugin.ts

```diff
@@ -15,14 +15,35 @@ import { BioAuthSaveParamsDto } from '../../remote/dto/bio.auth.save.params.dto'
 import { BioAuthSaveDto } from '../../remote/dto/bio.auth.save.dto';
 
 export class FingerprintPlugin implements IPlugin {
+    private readonly isExternal: boolean;
 
     /**
      * We pass in the parent class as context so we can access the sms module
      */
     constructor(
         private readonly bioAuthService: IBioAuthService,
         private readonly externalIdDbGateway: ExternalIdDbGateway
-    ) { }
+    ) {
+        this.isExternal = process.env.EXTERNAL_BIO_AUTH === 'true';
+    }
+
+    private restrictToInternal(operation: string) {
+        if (this.isExternal) {
+            throw new ProtocolException(ProtocolErrorCode.NOT_IMPLEMENTED, `External Bio Auth implementations do not support ${operation}`);
+        }
+    }
+
+    private async fingerprintQualityCheck(e: any, agentIds: string): Promise<any> {
```

Comment:
is it possible to type e?  if not can we give it better name to make it clearer as to what e is?